CMakeLists.txt

```diff
@@ -0,0 +1,96 @@
+    message(WARNING "Configured outside of scikit-build-core. Library installation is disabled.")
+    set(CLP_FFI_PY_INSTALL_LIBS OFF CACHE BOOL
+            "Disable library installation when building outside of scikit-build-core." FORCE)
+    set(CLP_FFI_PY "clp-ffi-py-local-build" CACHE STRING "Use local build name." FORCE)
+    set(CLP_FFI_PY_VERSION "0.0.12" CACHE STRING "Use local build version." FORCE)
+    # Enable compile commands by default if the generator supports it.
+    if (NOT CMAKE_EXPORT_COMPILE_COMMANDS AND CMAKE_GENERATOR MATCHES "Ninja|Unix Makefiles")
+        set(CMAKE_EXPORT_COMPILE_COMMANDS ON CACHE BOOL
+                "Enable/Disable output of compile commands during generation." FORCE)
+    endif()
+endif()
+
+project(${CLP_FFI_PY}
+    LANGUAGES CXX C
+    VERSION ${CLP_FFI_PY_VERSION}
+)
+
+# C-extension modules will be compiled as shared libraries. To enable dynamic linking, all libraries
+# (including the static ones) must be compiled with position-independent-code option.
+set(CMAKE_POSITION_INDEPENDENT_CODE ON CACHE BOOL "Compile libraries using PIC.")
+
+# Find Python interpreter, compiler and development environment.
+find_package(Python COMPONENTS Interpreter Development.Module REQUIRED)
+
+set(CPP_SRC_ROOT "src" CACHE STRING "Path to the root of all C++ source code")
+set(CLP_FFI_PY_SRC_ROOT "${CPP_SRC_ROOT}/clp_ffi_py" CACHE STRING
+        "Path to the root of clp-ffi-py's C++ source code")
+set(CLP_CORE_ROOT "src/clp/components/core" CACHE STRING "Path to CLP core's root")
+set(CLP_SRC_ROOT "${CLP_CORE_ROOT}/src/clp" CACHE STRING "Path to clp's source code")
+
+add_subdirectory(${CLP_SRC_ROOT}/string_utils)
+
+set(CLP_FFI_PY_LIB_IR "native" CACHE STRING "The name of IR native library")
+python_add_library(${CLP_FFI_PY_LIB_IR} MODULE WITH_SOABI)
+
+target_sources(${CLP_FFI_PY_LIB_IR}
+    PRIVATE
+    ${CLP_FFI_PY_SRC_ROOT}/ir/native/deserialization_methods.cpp
+    ${CLP_FFI_PY_SRC_ROOT}/ir/native/Metadata.cpp
+    ${CLP_FFI_PY_SRC_ROOT}/ir/native/PyDeserializer.cpp
+    ${CLP_FFI_PY_SRC_ROOT}/ir/native/PyDeserializerBuffer.cpp
+    ${CLP_FFI_PY_SRC_ROOT}/ir/native/PyFourByteSerializer.cpp
+    ${CLP_FFI_PY_SRC_ROOT}/ir/native/PyLogEvent.cpp
+    ${CLP_FFI_PY_SRC_ROOT}/ir/native/PyMetadata.cpp
+    ${CLP_FFI_PY_SRC_ROOT}/ir/native/PyQuery.cpp
+    ${CLP_FFI_PY_SRC_ROOT}/ir/native/Query.cpp
+    ${CLP_FFI_PY_SRC_ROOT}/ir/native/serialization_methods.cpp
+    ${CLP_FFI_PY_SRC_ROOT}/modules/ir_native.cpp
+    ${CLP_FFI_PY_SRC_ROOT}/Py_utils.cpp
+    ${CLP_FFI_PY_SRC_ROOT}/utils.cpp
+    ${CLP_SRC_ROOT}/BufferReader.cpp
+    ${CLP_SRC_ROOT}/ffi/ir_stream/decoding_methods.cpp
+    ${CLP_SRC_ROOT}/ffi/ir_stream/encoding_methods.cpp
+    ${CLP_SRC_ROOT}/ffi/ir_stream/utils.cpp
+    ${CLP_SRC_ROOT}/ffi/encoding_methods.cpp
+    ${CLP_SRC_ROOT}/ir/parsing.cpp
+    ${CLP_SRC_ROOT}/ReaderInterface.cpp
+)
+
+target_compile_features(${CLP_FFI_PY_LIB_IR}
+    PRIVATE
+    cxx_std_20
+)

```

Comment:
Would it be better to place this before the source files / includes definitions? (probably right after `python_add_library`

Reply:
Sure, make sense